### PR TITLE
Add X-Xss-Protection headerValue to XML config

### PR DESCRIPTION
**Pull request by Kehrlann**
See gh-11908

### CODE REVIEW

File: config/src/main/resources/org/springframework/security/config/spring-security-5.8.rnc [hunk 1]
```diff
@@ -1298,6 +1298,9 @@ xss-protection.attlist &=
 xss-protection.attlist &=
 	## Add mode=block to the header or not, default is on.
 	attribute block {xsd:boolean}?
+xss-protection.attlist &=
+	## Specify the value for the X-Xss-Protection header. When set, overrides both enabled and block attributes.
+	attribute header-value {"0","1","1\;mode=block"}?
```
Comment (sjohnr) [inline config/src/main/resources/org/springframework/security/config/spring-security-5.8.rnc:1303]: My IDE (IntelliJ) complains about this line, with

    group of "string" or "data" element

Is this the correct change to make?

**Review by sjohnr — CHANGES_REQUESTED** [review]
Thanks @Kehrlann! See inline comment below.

File: config/src/main/resources/org/springframework/security/config/spring-security-5.8.rnc (hunk 1 above)
Comment (Kehrlann) [reply config/src/main/resources/org/springframework/security/config/spring-security-5.8.rnc]: Good catch, thanks!

Changed to `"0" | "1" | "1; mode=block"`, it generates a very similar XSD but IntelliJ doesn't complain.

Note, I am no RELAX NG expert. I "made it work" but I don't really understand how...

**Comment by sjohnr** [comment]
Merged via 0e215a21ad491534b6615760b3b02099d978c8fe. I regenerated the 5.8 xsd, as it was out of sync, and added c98de7af2f6328468ae3303a9320d19c1649a9ab in 6.0 for the rnc/xsd after the merge.

**Pull request closed without merging**